### PR TITLE
fixed code smell at applicant-tasklist-utils

### PR DESCRIPTION
**Pull request by SasankAchyuta9**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/common/summary/utils.ts [hunk 1]
```diff
@@ -88,15 +88,18 @@ export const summaryList = (
   const summaryData: SummaryListRow[] = [];
   for (const key in keys) {
     const keyLabel = keys[key];
+    const getSelectedprivaedet = userCase[key] + getSelectedPrivateDetails(userCase);
+    const setkey = key1 => {
+      if (key1 === 'startAlternative' && userCase[key1] !== 'undefined') {
```
Comment (vivek-sekhar) [inline src/main/steps/common/summary/utils.ts:93]: is undefined of data type string?

File: src/main/steps/common/summary/utils.ts [hunk 2]
```diff
@@ -131,11 +134,9 @@ export const summaryCaseList = (
         caseUrl = RESPONDENT_TASK_LIST_URL + '/' + id;
       }
     } else if (userCase.caseTypeOfApplication === 'FL401') {
-      if (!isRespondent) {
-        caseUrl = APPLICANT_TASK_LIST_URL + '/' + id;
-      } else {
-        caseUrl = RESPONDENT_TASK_LIST_URL + '/' + id;
-      }
+      const App_ID = APPLICANT_TASK_LIST_URL + '/' + id;
```
Comment (vivek-sekhar) [inline src/main/steps/common/summary/utils.ts:137]: Can you please use the helper I have put in the framework for string concat?

File: src/main/steps/index.ts [hunk 3]
```diff
@@ -143,7 +102,8 @@ const getStepsWithContent = (sequence: Step[], subDir = ''): StepWithContent[] =
   if (sequence?.length) {
     for (const step of sequence) {
       const { url } = parseUrl(step.url);
-      const stepDir = `${dir}${url.startsWith(subDir) ? url : `${subDir}${url}`}`;
+      const subdirurl = url.startsWith(subDir) ? url : `${subDir}${url}`;
```
Comment (vivek-sekhar) [inline src/main/steps/index.ts:105]: What is the significance of these lines?

File: src/main/steps/respondent/respondentcaseSequence.ts [hunk 4]
```diff
@@ -81,17 +78,6 @@ export const respondentCaseSequence: Step[] = [
     showInSection: Sections.AboutRespondentCase,
     getNextStep: () => RESPONDENT_TASK_LIST_URL,
   },
-
-  {
```
Comment (vivek-sekhar) [inline src/main/steps/respondent/respondentcaseSequence.ts:85]: Where are these steps moved now?

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/respondent/respondentcaseSequence.ts (hunk 4 above)
Comment (SasankAchyuta9) [reply src/main/steps/respondent/respondentcaseSequence.ts:85]: These are part of response case sequence now. These occured due to duplication.

File: src/main/steps/index.ts (hunk 3 above)
Comment (SasankAchyuta9) [reply src/main/steps/index.ts:105]: Its actually came as a part of sonar  issue . And in the suggestions it showed to make literals with conditions  into two parts  .

File: src/main/steps/common/summary/utils.ts (hunk 1 above)
Comment (SasankAchyuta9) [reply src/main/steps/common/summary/utils.ts:93]: This is part of old code itself.I did not made any change on making that as string type.

**Pull request merged**